src/lib/cozy-client/reducer.js

```diff
@@ -198,6 +198,7 @@ const collection = (state = collectionInitialState, action) => {
       const response = action.response
       return {
         ...state,
+        type: state.type || action.doctype,
```

Comment:
Why this? Could you please give me some context?

Reply:
If you send a receive data with no fetch beforehand, the created collection has no doctype.

Reply:
I got that, but why (and where) is a RECEIVE_DATA dispatched with no FETCH_COLLECTION beforehand? Is it in your app code?

Reply:
@goldoraf in case of, I edited my message as you were writing your reply ^

Reply:
OK. You shouldn't have to do that kind of hack in your components' tests. As I understand it, what you want is to test the UI behavior of your component, is it really necessary to manipulate the "data layer" in order to test your component (it shouldn't)? Couldn't you test the "unconnected" version of the component?

Reply:
> Couldn't you test the "unconnected" version of the component?

Something like : 

```jsx
const UnconnectedReimbursements = function ({ bills }) {
  return ...
}

export UnconnectedReimbursements
export default cozyConnect(UnconnectedReimbursements)
```

and then in my tests do

```jsx
const bills =  data: fixtures['io.cozy.bills']
const dom = render(<UnconnectedReimbursements bills={{ data: bills, fetchStatus: 'loaded' }}, domNode)
expect(dom.querySelector('.bill').length).toBe(bills.length)
```

Reply:
Implemented in https://gitlab.cozycloud.cc/labs/cozy-sante/merge_requests/45/diffs?commit_id=f7cf11d385b10c5af4d5664a36942e6e9d17b5d1

I removed my change on the reducer 👍 Thanks !